core/wiki/macros/toolbar.tid

```diff
@@ -0,0 +1,73 @@
+title: $:/core/macros/toolbar
+tags: $:/tags/Macro
+
+\define toolbar-filter-display(tag, config)
+<$set name="config" filter="[[$config$]match[]]" 
```

Comment:
We are moving away from using macro parameters via textual substitution (ie `$foo$` syntax) because the technique is brittle to parameter values containing quotes or square brackets.

The preferred alternative is to use the `__parameterName__` syntax, whereby each macro parameter is available as a variable with the same name wrapped in double underscores.

So, the line above could be written:

```
<$set name="config" filter="[<__config__>match[]]"
```

Reply:
Done. I thought i did it.

There are more textual substitution but when filters are declared as variables/macros that alternative doesn't work (for me).